riscv-c-api.md

```diff
@@ -172,5 +172,48 @@ For example:
 
 ## Intrinsic Functions
 
-Do we really have none of these?  I can't figure out
-`gcc/gcc/config/riscv/riscv-builtins.c`...
+Intrinsic functions (or intrinsics or built-ins) are expanded into instruction sequences by compilers.
+They typically provide access to functionality that is otherwise not synthesizable by compilers.
+Some intrinsics expand to different code sequences depending on the available instructions from the enabled ISA extensions.
+
+Compilers typically come with their own architecture-independent intrinsics (e.g. synchronization primitives, byte-swap, etc.).
+The RISC-V compiler backend can define additional target-specific intrinsics.
+Providing functionality via architecture-independent intrinsics is the preferred method, as it improves code portability.
+
+Some intrinsics are only available if a particular header file is included.
+RISC-V header files that enable intrinsics require the prefix `riscv_` (e.g. `riscv_vector.h` or `riscv_crypto.h`).
```

Comment:
Is there any rules about the header file naming if header file required?

Reply:
Short answer: no rules beyond the prefix `riscv_`.

The reason is that this guide cannot foresee which grouping makes sense.
E.g. vector crypto instructions might be gated by `riscv_vector.h`, `riscv_vector_crypto.h`, or `riscv_crypto.h`. All of them would be valid candidates (maybe even multiple options), so this will have to be decided on a case-by-case basis.

Reply:
Got it, and it will be good to have same filename naming rule for different toolchains.